src/main/java/com/passbook/sparkeighteen/peristence/repository/UserRepository.java

```diff
@@ -0,0 +1,10 @@
+package com.passbook.sparkeighteen.peristence.repository;
+
+import com.passbook.sparkeighteen.peristence.entity.UserEntity;
+import org.springframework.data.jpa.repository.JpaRepository;
+
+import java.util.Optional;
+
+public interface UserRepository  extends JpaRepository<UserEntity, Integer> {
+    Optional<UserEntity> findByEmail(String email);
```

Comment:
Why do we use optional let me know.

Reply:
**Optional**  is java 8 feature and it is a container object which may or may not contain a non-null value.If a value is present, isPresent() will return true and get() will return the value